### ISSUE TITLE
false positive:  --manifest-path 不支持类似 e1000-driver 仓库的二级 .cargo 配置文件方式

### ISSUE THREAD

**Issue by zjp-CN** [os-checker/os-checker#22]
<img src="https://github.com/user-attachments/assets/86d36882-567e-4cb8-b09d-ace09bb9a761" height="220px"><img src="https://github.com/user-attachments/assets/266e722d-ca03-4cfa-bf4c-f66fe9af6b7a" height="220px">

e1000-driver 仓库代码：https://github.com/elliott10/e1000-driver/blob/c4aba40d5fd3ac660a0c1be2feaa3f897f1042ae/examples/.cargo/config

解释： e1000-driver 具有两个 packages
* 对于 p1 e1000-driver，无 .cargo 配置文件，意味着默认以主机工具链作为编译目标来检查
* 对于 p2 e1000-driver-test，含 .cargo 配置文件，以指定的 riscv64gc-unknown-none-elf 为目标进行编译

那么，在 `/ci/e1000-driver` 目录下运行 `cargo clippy --no-deps --message-format=json --manifest-path /ci/e1000-driver/examples/Cargo.toml` 时，cargo 并不查看 `/ci/e1000-driver/examples/.cargo/config` 文件，以主机目标编译，造成一系列 ``error: invalid register `x10`: unknown register`` 未知寄存器的 rustc 错误。

```rust
2024-08-11T05:20:59.550649Z DEBUG os_checker::layout: layout=Layout {
    repo_root: "e1000-driver",
    cargo_tomls: [
        "e1000-driver/Cargo.toml",
        "e1000-driver/examples/Cargo.toml",
    ],
    workspaces: Workspaces {
        [0] root: "./",
        [0] root.members: [
            "e1000-driver",
        ],
        [1] root: "./examples",
        [1] root.members: [
            "e1000-driver-test",
        ],
    },
}
...
2024-08-11T05:20:59.556922Z DEBUG os_checker::repo::cmd: expr=Cmd(["cargo", "clippy", "--no-deps", "--message-format=json", "--manifest-path", "/ci/e1000-driver/examples/Cargo.toml"])
```

解决方式：不应该指定 `--manifest-path` 来运行，而是进入 `--manifest-path` 所在的目录，运行 `cargo clippy --no-deps --message-format=json`，让 Cargo 自己检查 .cargo 配置。这可能需要应用到所有基于 cargo 的检查工具，现在和将来。

~~不确定这是不是 Cargo 的预期行为，似乎应该制作一个 MRE 询问一下。~~ 应该是 Cargo 预期的行为 ([MRE](https://github.com/os-checker/MRE-cargo-manifest)) ，因为 Cargo 检测 .cargo 配置并不考虑 --manifest-path 参数传入 ([ref](https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure))，所以需要进入那个被检查目录才行。

**Comment by zjp-CN** [comment]
由 https://github.com/os-checker/os-checker/commit/e9cf7e1749a72a36eeb58a92ef4abccbc3111e4d 成功解决该问题：

![截图_20240811185740](https://github.com/user-attachments/assets/93080e98-7ee3-4534-8d73-bc0338ee44ca)
